src/domain/community/virtualContributor/vcSettingsPage/VirtualContributorForm.tsx

```diff
@@ -0,0 +1,186 @@
+import VisualUpload from '../../../../core/ui/upload/VisualUpload/VisualUpload';
+import Section from '../../../../core/ui/content/deprecated/Section';
+import Gutters from '../../../../core/ui/grid/Gutters';
+import useLoadingState from '../../../shared/utils/useLoadingState';
+import { Actions } from '../../../../core/ui/actions/Actions';
+import { LoadingButton } from '@mui/lab';
+import { TagsetSegment } from '../../../platform/admin/components/Common/TagsetSegment';
+import { UpdateTagset } from '../../../common/profile/Profile';
+import FormikInputField from '../../../../core/ui/forms/FormikInputField/FormikInputField';
+import { theme } from '../../../../core/ui/themes/default/Theme';
+
+interface VirtualContributorProps {
+  id: string;
+  nameID: string;
+  profile: {
+    id: string;
+    displayName: string;
+    description?: string;
+    tagline: string;
+    tagsets?: Tagset[] | undefined;
+    url: string;
+    avatar?: Visual | undefined;
+  };
+}
+
+interface VirtualContributorFromProps {
+  name: string;
+  nameID: string;
+  description: string;
+  tagline: string;
+  tagsets?: Tagset[];
+  avatar: Visual | undefined;
+}
+
+interface Props {
+  virtualContributor: VirtualContributorProps;
+  avatar: Visual | undefined;
+  onSave?: (virtualContributor: UpdateVirtualContributorInput) => void;
+  title?: string;
+}
+
+export const VirtualContributorForm: FC<Props> = ({
+  virtualContributor: currentVirtualContributor,
+  avatar,
+  onSave,
+}) => {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+
+  const {
+    nameID,
+    profile: { displayName, description, tagline, tagsets },
+  } = currentVirtualContributor;
+
+  const initialValues: VirtualContributorFromProps = {
+    name: displayName,
+    nameID: nameID,
+    description: description ?? '',
+    tagline: tagline,
+    avatar: avatar,
+    tagsets: tagsets,
+  };
+
+  const validationSchema = yup.object().shape({
+    name: nameSegmentSchema.fields?.name || yup.string(),
+    nameID: nameSegmentSchema.fields?.nameID || yup.string(),
+    description: profileSegmentSchema.fields?.description || yup.string(),
+  });
+
+  const getUpdatedTagsets = (updatedTagsets: Tagset[]) => {
+    const result: UpdateTagset[] = [];
+    updatedTagsets.forEach(updatedTagset => {
+      const originalTagset = tagsets?.find(value => value.name === updatedTagset.name);
+      if (originalTagset) result.push({ ...originalTagset, tags: updatedTagset.tags });
+    });
+
+    return result;
+  };
+
+  const [handleSubmit, loading] = useLoadingState(async (values: VirtualContributorFromProps) => {
+    const { tagsets, description, tagline, name, ...otherData } = values;
+    const updatedTagsets = getUpdatedTagsets(tagsets || []);
+
+    const virtualContributor = {
+      ID: currentVirtualContributor.id,
+      profileData: {
+        displayName: name,
+        description,
+        tagline,
+        tagsets: updatedTagsets.map(r => ({
+          ID: r.id,
```

Comment:
is this `id: undefined` required? you're creating a new object...
also I would call the variable `tagset`

Reply:
I'm displaying the tagsets, there's no default tagset. I could use some help on switching from default keywords and capabilities to just tags.